pkg/virt-config/virt-config.go

```diff
@@ -130,6 +130,10 @@ func (c *ClusterConfig) GetResourceVersion() string {
 }
 
 func (c *ClusterConfig) GetMachineType(arch string) string {
+	if c.GetConfig().MachineType != "" {
+		return c.GetConfig().MachineType
+	}
+
```

Comment:
Shouldn't we give precedence  to the `ArchitectureConfiguration` and use the `MachineType` as fallback? IOW, move this under the `default` case of the `switch`? Thanks